internal/services/machinelearning/machine_learning_workspace_network_outbound_rule_fqdn_resource.go

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package machinelearning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2024-04-01/managednetwork"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2024-04-01/workspaces"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = WorkspaceNetworkOutboundRuleFqdn{}
+
+type machineLearningWorkspaceOutboundRuleFqdnModel struct {
+	Name        string `tfschema:"name"`
+	WorkspaceId string `tfschema:"workspace_id"`
+	Destination string `tfschema:"destination"`
+}
+
+type WorkspaceNetworkOutboundRuleFqdn struct{}
+
+var _ sdk.Resource = WorkspaceNetworkOutboundRuleFqdn{}
+
+func (r WorkspaceNetworkOutboundRuleFqdn) ResourceType() string {
+	return "azurerm_machine_learning_workspace_network_outbound_rule_fqdn"
+}
+
+func (r WorkspaceNetworkOutboundRuleFqdn) ModelObject() interface{} {
+	return &machineLearningWorkspaceOutboundRuleFqdnModel{}
+}
+
+func (r WorkspaceNetworkOutboundRuleFqdn) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return managednetwork.ValidateOutboundRuleID
+}
+
+func (r WorkspaceNetworkOutboundRuleFqdn) Arguments() map[string]*pluginsdk.Schema {
+	arguments := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"workspace_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: managednetwork.ValidateWorkspaceID,
+		},
+
+		"destination": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
should we be validating this is a FQDN?

Reply:
There is no restriction on FQDN in the portal, as long as it is a string, checking that the string should not be empty is enough